Add compile time guard for fullscreen video dismissal gestures

grorg opened the pull request:
#### f0703be2f94ac0255a2231e2eed54bc59b575c52
<pre>
Add compile time guard for fullscreen video dismissal gestures
<a href="https://bugs.webkit.org/show_bug.cgi?id=242222">https://bugs.webkit.org/show_bug.cgi?id=242222</a>

Reviewed by Antoine Quint and Devin Rousso.

Add ENABLE(FULLSCREEN_DISMISSAL_GESTURES), which is enabled
on PLATFORM(IOS_FAMILY).

* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenWindowControllerIOS.mm:
(-[WKFullScreenWindowController enterFullScreen]):

Canonical link: <a href="https://commits.webkit.org/252074@main">https://commits.webkit.org/252074@main</a>
</pre>

dcrousso commented on Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenWindowControllerIOS.mm:
```diff
@@ -550,6 +550,7 @@ - (void)enterFullScreen
     [self _updateLocationInfo];
 
+#if HAVE(FULLSCREEN_DISMISSAL_GESTURES)
     _startDismissGestureRecognizer = adoptNS([[UISwipeGestureRecognizer alloc] initWithTarget:self action:@selector(_startToDismissFullscreenChanged:)]);
```
I don't think guarding this is enough.  There's lots of uses of these ivars in other methods (not to mention the ivars and methods themselves) that we probably want to guard too.

Also, what about `_interactiveDismissTransitionCoordinator`?

grorg replied:
There were a couple of other places. We didn't really need to guard the ivars or the gesture recognizer calls, but it is cleaner.

dcrousso commented on Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenWindowControllerIOS.mm:
```diff
@@ -550,6 +552,7 @@ - (void)enterFullScreen
     [self _updateLocationInfo];
 
+#if ENABLE(FULLSCREEN_DISMISSAL_GESTURES)
     _startDismissGestureRecognizer = adoptNS([[UISwipeGestureRecognizer alloc] initWithTarget:self action:@selector(_startToDismissFullscreenChanged:)]);
```
we probably could/should also guard `_startToDismissFullscreenChanged:` too while we're doing the rest :)

grorg committed:
Add compile time guard for fullscreen video dismissal gestures
https://bugs.webkit.org/show_bug.cgi?id=242222

Reviewed by Antoine Quint and Devin Rousso.

Add ENABLE(FULLSCREEN_DISMISSAL_GESTURES), which is enabled
on PLATFORM(IOS_FAMILY).

* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenWindowControllerIOS.mm:
(-[WKFullScreenWindowController enterFullScreen]):

Canonical link: https://commits.webkit.org/252074@main
